apps/web/pages/settings/my-account/profile.tsx

```diff
@@ -203,133 +179,100 @@ const ProfileView = () => {
     [ErrorCode.InternalServerError]: `${t("something_went_wrong")} ${t("please_try_again_and_contact_us")}`,
     [ErrorCode.ThirdPartyIdentityProviderEnabled]: t("account_created_with_identity_provider"),
   };
-  const onSuccessfulUsernameUpdate = async () => {
-    showToast(t("settings_updated_successfully"), "success");
-    await utils.viewer.me.invalidate();
-  };
-
-  const onErrorInUsernameUpdate = () => {
-    showToast(t("error_updating_settings"), "error");
-  };
 
-  if (isLoading || !user)
+  if (isLoading || !user || isLoadingAvatar || !avatar)
     return <SkeletonLoader title={t("profile")} description={t("profile_description")} />;
 
-  const isDisabled = isSubmitting || !isDirty;
+  const defaultValues = {
+    username: user.username || "",
+    avatar: avatar.avatar || "",
+    name: user.name || "",
+    email: user.email || "",
+    bio: user.bio || "",
+  };
 
   return (
     <>
       <Meta title={t("profile")} description={t("profile_description", { appName: APP_NAME })} />
-      <Form
-        form={formMethods}
-        handleSubmit={(values) => {
-          if (values.email !== user?.email && isCALIdentityProviver) {
+      <ProfileForm
+        key={JSON.stringify(defaultValues)}
+        defaultValues={defaultValues}
+        onSubmit={(values) => {
+          if (values.email !== user.email && isCALIdentityProviver) {
+            setTempFormValues(values);
             setConfirmPasswordOpen(true);
           } else {
             mutation.mutate(values);
           }
-        }}>
-        <div className="flex items-center">
-          <Controller
-            control={formMethods.control}
-            name="avatar"
-            render={({ field: { value } }) => (
-              <>
-                <Avatar alt="" imageSrc={value} gravatarFallbackMd5={emailMd5} size="lg" />
-                <div className="ml-4">
-                  <ImageUploader
-                    target="avatar"
-                    id="avatar-upload"
-                    buttonMsg={t("change_avatar")}
-                    handleAvatarChange={(newAvatar) => {
-                      formMethods.setValue("avatar", newAvatar, { shouldDirty: true });
-                    }}
-                    imageSrc={value}
-                  />
-                </div>
-              </>
+        }}
+        extraField={
+          <div className="mt-8">
+            <UsernameAvailabilityField
+              user={user}
+              onSuccessMutation={async () => {
+                showToast(t("settings_updated_successfully"), "success");
+                await utils.viewer.me.invalidate();
+              }}
+              onErrorMutation={() => {
+                showToast(t("error_updating_settings"), "error");
+              }}
+            />
+          </div>
```

Comment:
Felt wrong having that much nested forms so this an attempt to cleanup and reorganize. Far from perfect tho.